### DISCUSSION TITLE
Welcome to python-snap7 Discussions!

### DISCUSSION THREAD

**Discussion by gijzelaerr** [gijzelaerr/python-snap7#246] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by Octopusmode** [comment]
Good afternoon.
How do I read an Array of Real (or other types) into a list?

Here is an example with a data-block from which I want to read data.
![image](https://user-images.githubusercontent.com/26312857/145031149-1f9ea525-226f-4d0b-a5b7-7ddb46587808.png)

I'm new to Simatic.
I understand how to interact with single bytes, integers, strings, but I don't understand how to interact with arrays.

**Reply by swamper123** [reply]
Hey ho,

first: Next time, open a new Discussion Thread [here](https://github.com/gijzelaerr/python-snap7/discussions/new). ;)

You know the db, you know the startbyte and you know how many bytes are in that array. So you can read the whole array via `db_read()`.
Then you got the raw data of the array. To get a double value, you can use [get_real()](https://github.com/gijzelaerr/python-snap7/blob/b6481ec45bb6652288e991d6d90594b6bf2c64d0/snap7/util.py#L391) or be inspired to make your own function for arrays.

That should do the thing.

**Reply by Octopusmode** [reply]
Got it, I'll do it right next time.
I already found the answer to my question at https://python-snap7.readthedocs.io/en/latest/_modules/snap7/util.html

I used the **get_real()** function from there like you wrote.
Thanks for the help!

**Comment by vgalisson** [comment]
Hi ! I'm wondering where is the best place to share a problem (not related to https://github.com/gijzelaerr/python-snap7/discussions/330) and its solution to the community ? Should I post it in Discussions or open an issue instead ?

**Reply by swamper123** [reply]
I would say:

- If you got an issue/found a bug/... -> [Make an issue](https://github.com/gijzelaerr/python-snap7/issues/new).
--> Worst case is, that it may be converted to a discussion point.
- If you got a question/idea -> [A new Discussion](https://github.com/gijzelaerr/python-snap7/discussions/new)
- If you already coded a new feature -> [You can make a PR ](https://github.com/gijzelaerr/python-snap7/compare) and explain why it is cool
- If you got an issue/found a bug AND you got a solution for that ->  [Make an issue](https://github.com/gijzelaerr/python-snap7/issues/new), then [a PR ](https://github.com/gijzelaerr/python-snap7/compare) whit the reference to issue #XYZ. 

EDIT: Since THIS is a new question, you should have made a new Discussion. ;)
